centreon/src/App/ResourceConfiguration/Infrastructure/ApiPlatform/State/ListGlobalMacrosProvider.php

```diff
@@ -0,0 +1,96 @@
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace App\ResourceConfiguration\Infrastructure\ApiPlatform\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\Pagination\Pagination;
+use ApiPlatform\State\Pagination\TraversablePaginator;
+use ApiPlatform\State\ProviderInterface;
+use App\ResourceConfiguration\Domain\Aggregate\GlobalMacro;
+use App\ResourceConfiguration\Domain\Repository\GlobalMacroCriteria;
+use App\ResourceConfiguration\Domain\Repository\GlobalMacroRepository;
+use App\ResourceConfiguration\Infrastructure\ApiPlatform\Resource\GlobalMacroResource;
+use App\ResourceConfiguration\Infrastructure\ApiPlatform\Transformer\GlobalMacroTransformer;
+use App\Shared\Domain\Repository\Paginator;
+use App\Shared\Infrastructure\ApiPlatform\Transformer\TransformerInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+
+/**
+ * @implements ProviderInterface<GlobalMacroResource>
+ */
+final readonly class ListGlobalMacrosProvider implements ProviderInterface
+{
+    /**
+     * @param TransformerInterface<GlobalMacro,GlobalMacroResource> $transformer
+     */
+    public function __construct(
+        #[Autowire(service: GlobalMacroTransformer::class)]
+        private TransformerInterface $transformer,
+        private GlobalMacroRepository $repository,
+        private Pagination $pagination,
+    ) {
+    }
+
+    /**
+     * @return iterable<GlobalMacroResource>
+     */
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): iterable
+    {
+        $criteria = new GlobalMacroCriteria();
+        if ($this->pagination->isEnabled($operation, $context)) {
+            $criteria = $criteria->withPagination(
+                $this->pagination->getPage($context),
+                $this->pagination->getLimit($operation, $context)
+            );
+        }
+        if (
+            isset($context['filters'])
+            && is_array($context['filters'])
+            && isset($context['filters']['name'])
+            && is_array($context['filters']['name'])
+        ) {
+            if (isset($context['filters']['name']['lk']) && is_string($context['filters']['name']['lk'])) {
+                $criteria = $criteria->withOperator('lk')->withName($context['filters']['name']['lk']);
+            } elseif (isset($context['filters']['name']['eq']) && is_string($context['filters']['name']['eq'])) {
+                $criteria = $criteria->withOperator('eq')->withName($context['filters']['name']['eq']);
+            }
```

Comment:
You probably don't want to add new elseif in this file (and all the others) for each new supported operators.

```suggestion
            foreach (['lk', 'eq'] as $op) {
                if (isset($nameFilter[$op]) && is_string($nameFilter[$op])) {
                    $criteria = $criteria->withOperator($op)->withName($nameFilter[$op]);
                    break;
                }
            }
```

The array of operators may be stored in a const of this class for now, at a more global level later on WDYT?

Reply:
The array could be defined in the Criteria Class as the operators are used by this class